README
AnyEvent-Redis-RipeRedis version 1.60
======================

INSTALLATION

To install this module type the following:

  perl Makefile.PL
  make
  make test
  make install

DESCRIPTION

MODULE IS DEPRECATED. Use AnyEvent::RipeRedis instead. The interface of
AnyEvent::RipeRedis has several differences from interface of
AnyEvent::Redis::RipeRedis. For more information see documentation.

AnyEvent::Redis::RipeRedis is the flexible non-blocking Redis client with
reconnect feature. The client supports subscriptions, transactions and
connection via UNIX-socket.

Requires Redis 1.2 or higher and any supported event loop.

AUTHOR

Eugene Ponizovsky, <ponizovsky@gmail.com>

Special thanks

* Alexey Shrub
* Vadim Vlasov
* Konstantin Uvarin
* Ivan Kruglov

COPYRIGHT AND LICENSE

Copyright (c) 2012-2016, Eugene Ponizovsky, <ponizovsky@gmail.com>.
All rights reserved.

This module is free software; you can redistribute it and/or modify it under
the same terms as Perl itself.


Changes
Revision history for AnyEvent::Redis::RipeRedis.

1.60 Mon Jul 26 15:03:25 MSK 2016
  - Module is deprecated. Use AnyEvent::RipeRedis instead. Made related changes
    in POD.
  - Slight refactoring and cosmetic changes.

1.52 Mon Jun 27 11:25:35 MSK 2016
  - BUGFIX Added error code mapping for new error prefixes.

1.50 Mon Jun 20 15:20:45 MSK 2016
  - FEATURE Improved syntax of calling of multiple word commands.
  - FEATURE Added new constructor parameter "min_reconnect_interval" and
    corresponding accessor.
  - FEATURE Added new error codes.
  - Slight refactoring of unit tests.

1.48 Sat Jun 27 15:31:24 MSK 2015
  - FEATURE Improved subscription mechanism. Now if connection was lost when
    client has active subscriptions, the 'on_error' or 'on_reply' callback,
    specified in appropriate '(p)subscribe' method, is called.
  - Slight refactoring.
  - Minor changes in POD.

1.46 Sat Mar 7 15:37:54 MSK 2015
  - FEATURE: Added new error code E_BUSY_KEY.
  - Module Test::RedisServer was copied to testing environment and was refactored.
    Now it has name Test::RedisRunner and doesn't require Any::Moose.
  - Slight refactoring.
  - Minor changes in POD.

1.44 Sat Nov 1 19:35:00 MSK 2014
  - FEATURE: Method info() now parsing information and statistics about the
    server. The result is passed to 'on_done' or 'on_reply' callback as a hash
    reference.
  - Minor internal changes.
  - Improved unit tests.

1.42 Tue Sep 9 18:30:00 MSK 2014
  - FEATURE: added constructor parameter 'handle_params'.
  - Constructor parameters 'autocork' and 'linger' now are deprecated.
  - Changed parsing logic of replies from redis-server.
  - Large refactoring.
  - Improved POD.
  - Improved unit tests.

1.40 Mon May 19 10:45:00 MSK 2014
  - FEATURE: added constructor parameter 'linger'. Thanks to Ivan Kruglov.
  - Slight internal changes.

1.38 Sun Jan 26 19:15:00 MSK 2014
  - FEATURE: added 'autocork' behavior, which can be enabled or disabled.
  - BUGFIX: fixed conversion to lower case of command keyword.
  - Minor internal changes.
  - Minor changes in POD.

1.36 Mon Jan 20 19:15:00 MSK 2014
  - Accepted new versioning convention.
  - Unit tests improved.
  - Minor internal changes.
  - Minor changes in POD.

1.346 Wed Jan 15 01:45:00 MSK 2014
  - Copyright updated.

1.345 Wed Jan 15 01:25:00 MSK 2014
  - Command keyword is converted to lowercase.

1.344 Wed Jan 15 00:45:00 MSK 2014
  - Minor changes.

1.343 Wed Jan 15 00:40:00 MSK 2014
  - Several changes in unit tests.

1.342 Wed Jan 15 00:30:00 MSK 2014
  - Fix in unit tests.

1.341 Tue Jan 14 22:15:00 MSK 2014
  - BUGFIX: Now, selected_database() method returns '0' if 'database' parameter
    was not specified in constructor.

1.340 Tue Jan 14 21:00:00 MSK 2014
  - FEATURE: Added selected_database() method.
  - BUGFIX: Now, client remembers last selected database after reconnection.
  - Minor changes in POD.
  - Required version of ExtUtils::MakeMaker decremented back to 6.64.
  - Several internal changes.

1.321 Sun Jan 12 14:30:00 MSK 2014
  - Required version of Test::More decremented to 0.98.

1.320 Sat Jan 11 23:40:00 MSK 2014
  - On premature connection close, the client additionally calls the common
    'on_error' callback.
  - The premature connection close or the premature destroy of the client object
    produce different warnings.
  - Internal API improved.
  - Unit tests improved.
  - Minor changes in POD.

1.314 Tue Dec 24 19:10:00 MSK 2013
  - Minor fix in unit tests.

1.312 Tue Dec 24 18:00:00 MSK 2013
  - Several changes in POD.
  - Several changes in tests.

1.311 Fri Dec 20 00:30:00 MSK 2013
  - Several fixes in POD.
  - Cosmetic changes.

1.310 Wed Dec 18 19:10:00 MSK 2013
  - FEATURE: Added error codes based on error prefixes.
  - Improved parsing of error prefixes.
  - Minor changes in POD.
  - Cosmetic changes in example scripts.

1.303 Tue Dec 17 23:00:00 MSK 2013
  - Added version numbers to dependencies in BUILD_REQUIRES section in Makefile.PL.
  - Increased several timeouts in unit tests.

1.302 Tue Dec 17 01:20:00 MSK 2013
  - Fixed typo in POD.

1.301 Tue Dec 17 01:07:00 MSK 2013
  - Some fixes in POD.

1.300 Mon Dec 16 19:50:00 MSK 2013
  - FEATURE: Now, in command methods can be specified single callback, which
    is called in both cases: when operation was completed successfully and when
    some error occurred.
  - Improved POD.
  - Improved unit tests.
  - Another wave of refactoring.

1.254 Mon Jun 9 17:44:55 MSK 2013
  - Minor changes in POD.

1.253 Wed May 31 14:38:35 MSK 2013
  - BUGFIX: Fixed premature triggering of the 'rtimeout' timer in some cases.

1.252 Wed May 29 11:00:45 MSK 2013
  - BUGFIX: Fixed reading from the uninitialized read buffer in some cases.
  - Now, in tests empty TCP port is taken using module Net::EmptyPort.

1.251 Fri Apr 26 11:29:05 MSK 2013
  - Slight modification of examples in POD

1.250 Thu Apr 25 22:00:30 MSK 2013
  - Improved error handling for EXEC command and for Lua scripts, which returns
    errors in multi-bulk reply.
  - Word 'command' replaced on 'operation' in error messages.
  - Added test for memory leaks.
  - Improved other tests.
  - POD has been complemented.

1.244 Sat Apr 13 00:41:25 MSK 2013
  - Added CONFIGURE_REQUIRES section in Makefile.PL.
  - TEST_REQUIRES section renamed back to BUILD_REQUIRES.

1.243 Sat Apr 13 00:41:25 MSK 2013
  - Minor fixes in POD.

1.242 Fri Apr 12 16:54:10 MSK 2013
  - Increased minimal required version of Perl up to v5.8.0
  - Test 10-exceptions.t rewritten with using Test::Fatal
  - Edited POD.
  - Minor changes in examples.
  - Fixed dates in Changes file
  - Fixed year in copyright in README file

1.241 Wed Apr 10 07:35:25 MSK 2013
  - Removed Test::MockObject from dependencies.
  - Minor fix in POD.

1.240 Tue Apr 9 19:40:55 MSK 2013
  - FEATURE: Now, destroying of object can produce warnings if some commands
    were not completed.
  - BUGFIX: Fixed parsing of empty bulk reply.

Makefile.PL
use 5.008000;

use ExtUtils::MakeMaker;

WriteMakefile(
  NAME => 'AnyEvent::Redis::RipeRedis',
  VERSION_FROM => 'lib/AnyEvent/Redis/RipeRedis.pm', # finds $VERSION
  MIN_PERL_VERSION => '5.008000',
  PREREQ_PM => {
    'AnyEvent'     => 6.01,
    'Encode'       => 0,
    'Scalar::Util' => 0,
    'Digest::SHA'  => 0,
    'Carp'         => 0,
  },
  CONFIGURE_REQUIRES => {
    'ExtUtils::MakeMaker' => 6.64,
  },
  BUILD_REQUIRES => {
    'Test::More'        => 0.98,
    'Test::TCP'         => 2.12,
    'Test::Fatal'       => 0.013,
    'Time::HiRes'       => 0,
    'File::Temp'        => 0.19,
    'POSIX'             => 0,
    version             => 0.77,
  },
  META_MERGE => {
    resources => {
      homepage   => 'https://github.com/iph0/AnyEvent-Redis-RipeRedis',
      bugtracker => 'https://github.com/iph0/AnyEvent-Redis-RipeRedis/issues',
      repository => 'https://github.com/iph0/AnyEvent-Redis-RipeRedis',
      license    => 'http://dev.perl.org/licenses/',
    },
  },
  ABSTRACT_FROM => 'lib/AnyEvent/Redis/RipeRedis.pm',
  AUTHOR        => 'Eugene Ponizovsky <ponizovsky@gmail.com>',
  LICENSE       => 'perl',
);
